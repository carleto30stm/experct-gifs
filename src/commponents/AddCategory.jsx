import { useState } from "react"
import { PropTypes } from 'prop-types';

export const AddCategory = ({onNewCategory}) => {

    const [inputValue, setInputValue] = useState('');

    const onInputChange = ({target}) => {
        setInputValue(target.value);    
    }

    const onSubmit = (event) => {
        const newValue = inputValue.trim();
        event.preventDefault();

        if(newValue.length <= 1) return;

        onNewCategory(newValue);
        setInputValue('');
    
    }


  return (
    <form onSubmit={onSubmit}>
        <input 
            type="text"
            placeholder="Buscar Gif"
            value={inputValue}
            onChange={onInputChange}
        />
    </form>
  )
}

AddCategory.propTypes = {
    onNewCategory: PropTypes.func.isRequired
}   

