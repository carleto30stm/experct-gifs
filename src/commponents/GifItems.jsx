// crea una funcion suma

// eslint-disable-next-line react/prop-types
export const GifItems = ({title, url}) => (
  <div className="card">
    <img src={url} alt={title} />
    <p>{title}</p>
  </div>
);

