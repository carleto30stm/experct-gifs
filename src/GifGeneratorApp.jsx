import { useState } from "react"
import { AddCategory, GifGrid } from "./commponents";


const GifGeneratorApp = () => {

    const [categories, setCategories] = useState(['one', 'dragon'])
    const onAddCategory = (newCategory) => {
        newCategory = newCategory.toLowerCase().trim();
        
        if(categories.includes(newCategory)) return;

        setCategories([newCategory,...categories])
    }
    
  return (
    <>
        <div>GifGeneratorApp</div>
        
            <AddCategory onNewCategory={onAddCategory}/>
        {
            categories.map(category => (
                 <                    
                 GifGrid
                    key={category}
                    category={category}
                 />
            ))
        }

    </>
  )
}

export default GifGeneratorApp