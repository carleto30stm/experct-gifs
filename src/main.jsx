import { StrictMode } from 'react'
import { createRoot } from 'react-dom/client'
import GifGeneratorApp from './GifGeneratorApp'
import './styles.css'

createRoot(document.getElementById('root')).render(
  <StrictMode>
    <GifGeneratorApp />
  </StrictMode>,
)
