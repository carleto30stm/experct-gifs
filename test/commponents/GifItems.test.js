/* eslint-disable no-undef */
import { render, screen } from "@testing-library/react";
import { GifItems } from "../../src/commponents";


describe('Teste de componente GifItems', () => {
    const title = 'Saitama';
    const url = "https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=8a210d3e9rr4vzjgop42j2lc84yr3bl8jgv6pwrskicwv1tz&ep=v1_gifs_search&rid=giphy.gif&ct=g";

    // eslint-disable-next-line no-undef
    test('Debe de hacer match con el snapshot', () => {
      
        const { container } = render(<GifItems title={title} url={url} />);
        // eslint-disable-next-line no-undef
        expect(container).toMatchSnapshot();        
    });

    test('Debe de mostrar la imagen con el URL y el ALT indicado', () => {
       
        render(<GifItems title={title} url={url} />);
        const { src, alt } = screen.getByRole('img');

        expect(src).toBe(url);
        expect(alt).toBe(title);
    })
    test('Debe de mostrar el titulo en el componente', () => {
    
        render(<GifItems title={title} url={url} />);

        expect(screen.getByText(title)).toBeTruthy();
    })  
    
})