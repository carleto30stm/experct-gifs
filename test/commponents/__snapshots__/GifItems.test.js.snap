// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Teste de componente GifItems Debe de hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media2.giphy.com/media/VXJWhaO7afRe/giphy.gif?cid=8a210d3e9rr4vzjgop42j2lc84yr3bl8jgv6pwrskicwv1tz&ep=v1_gifs_search&rid=giphy.gif&ct=g"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
